azure/durable_functions/models/DurableOrchestrationContext.py

```diff
@@ -143,7 +143,7 @@ def _set_is_replaying(self, is_replaying: bool):
         """
         self._is_replaying = is_replaying
 
-    def call_activity(self, name: str, input_: Optional[Any] = None) -> TaskBase:
+    def call_activity(self, name: str | Callable, input_: Optional[Any] = None) -> TaskBase:
```

Comment:
We'll need to think about whether the parameter-name `name` is still the most appropriate now that we're supporting calling both via a `str` and via a function-reference. If making breaking changes came at no cost, I would simply rename this parameter to `activity` which seems like a more appropiate fit for both `str` and `Callable` types.

That opens a whole can of worms though so I want to be careful here. I could definitely justify keeping the parameter as is, but I'd like a third opinion.